proto/minder/v1/minder.proto

```diff
@@ -2290,33 +2332,78 @@ message Profile {
         // id is optional and use for updates to match upserts as well as read operations. It is ignored for creates.
         string id = 1;
         // entity is the entity to select.
-        string entity = 2;
+        string entity = 2 [
+            (buf.validate.field).string = {
+                pattern: "^[a-z]+(_[a-z]+)*$",
+                min_len: 1,
+                max_len: 200,
+            },
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
         // expr is the expression to select the entity.
-        string selector = 4;
+        string selector = 4 [
+            (buf.validate.field).string = {
+                pattern: "^[a-zA-Z_][a-zA-Z0-9_]*(?:\\.[a-zA-Z_][a-zA-Z0-9_]*)*(?:\\[\\d+\\])?$",
+                max_len: 200,
+            },
+            (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+        ];
         reserved 5;
         reserved "comment";
         // description is the human-readable description of the selector.
-        string description = 6;
+        string description = 6 [
+            (buf.validate.field).string = {
+                pattern: "[A-Za-z][-/.!?,:;[:word:] ]*",
```

Comment:
Why not `[[:punct:][:word:] ]*`?  Are there specific characters we're trying to exclude from the punctuation set?

(Also, why a leading letter?)

Reply:
Yes, `punct` seems to include < >  which we don't want right? 
As for the leading letter, I thought it feels better to establish some form of common formatting for when this value is visualised.